src/detect-parse.c

```diff
@@ -1904,6 +1905,11 @@ static int SigValidate(DetectEngineCtx *de_ctx, Signature *s)
         SCReturnInt(0);
     }
 
+    if (s->init_data->init_flags & SIG_FLAG_INIT_OVERFLOW) {
+        SCLogError("rule %u tries to use too many buffers", s->id);
+        SCReturnInt(0);
+    }
+
```

Comment:
Q: if there was a failure expanding the rule buffer array, could `s->init_data->curbuf->head` be NULL?
Asking to understand why this check wasn't added prior to the if statement before this.
Other than this, this patch makes sense to me. :)

Reply:
> I do not know. Why is it important ?
> 
> If `SigValidate` already returned that the signature is invalid for another reason, it is fine

I see. I had just thought we would want to say the exact the reason for invalid sig. If you think its fine either way, fine by me. Thanks :)

Reply:
There could be other reasons that come later and we skip them...